charts/posit-chronicle/templates/stateful-set.yaml

```diff
@@ -1,47 +1,65 @@
 ---
+{{- $root := . }}
 apiVersion: apps/v1
 kind: StatefulSet
 metadata:
-  name: chronicle-server
-  namespace: {{ $.Release.Namespace }}
+  name: {{ include "posit-chronicle.fullname" . }}
+  namespace: {{ include "posit-chronicle.namespace" . }}
+  labels:
+    {{- include "posit-chronicle.labels" . | nindent 4 }}
+  {{- with .Values.commonAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
 spec:
-  serviceName: chronicle-server
   replicas: {{ .Values.replicas }}
+  serviceName: {{ include "posit-chronicle.fullname" . }}
   selector:
     matchLabels:
-      {{- include "posit-chronicle.pod.selectorLabels" . | trim | nindent 6 }}
+      {{- include "posit-chronicle.selectorLabels" . | trim | nindent 6 }}
   template:
     metadata:
       labels:
-        {{- include "posit-chronicle.pod.labels" . | trim |  nindent 8 }}
-        {{- include "posit-chronicle.pod.selectorLabels" . | trim |  nindent 8 }}
+        {{- include "posit-chronicle.labels" . | trim |  nindent 8 }}
+        {{- with .Values.pod.labels }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
       annotations:
-        {{- include "posit-chronicle.pod.annotations" . | trim |  nindent 8 }}
+        checksum/config: {{ print .Values.config | sha256sum }}
+        {{- include "posit-chronicle.pod.annotations" . | trim | nindent 8 }}
     spec:
       {{- with .Values.pod.affinity }}
       affinity:
-        {{- toYaml . | nindent 8 }}
+        {{- tpl (toYaml .) $root | nindent 8 }}
       {{- end }}
       {{- with .Values.pod.nodeSelector }}
       nodeSelector:
-        {{- toYaml . | nindent 8 }}
+        {{- tpl (toYaml .) $root | nindent 8 }}
       {{- end }}
       {{- with .Values.pod.tolerations }}
       tolerations:
-        {{- toYaml . | nindent 8 }}
-      {{- end }}
-      {{- if .Values.serviceaccount.create }}
-      serviceAccountName: chronicle-sa
+      {{- tpl (toYaml .) $root | nindent 6 }}
       {{- end }}
+      serviceAccountName: {{ include "posit-chronicle.serviceAccountName" . }}
       containers:
-      - name: chronicle-server
-        image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
-        imagePullPolicy: {{ .Values.image.imagePullPolicy }}
+      - name: {{ .Chart.Name }}
+        {{- if .Values.image.sha }}
+        image: "{{ required "An image registry must be specified" .Values.image.registry }}/{{ required "An image repository must be specified" .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}@sha256:{{ .Values.image.sha }}"
+        {{- else }}
+        image: "{{ required "An image registry must be specified" .Values.image.registry }}/{{ required "An image repository must be specified" .Values.image.repository }}:{{ default .Chart.AppVersion .Values.image.tag }}"
+        {{- end}}
```

Comment:
Ok - I think this answers my earlier question about the image being customizeable.